### PR TITLE
Axios: Update add company functional test

### PR DESCRIPTION
**Pull request by tomkingston**
## Description of change

Adds functional tests for adding companies as part of the work to replace requests with axios.

The existing `add-company-spec.js` was quite difficult to debug as a result of excessive context nesting - it also depended on events from previous tests, which is an anti-pattern as described in the cypress docs (https://docs.cypress.io/guides/references/best-practices.html#Having-tests-rely-on-the-state-of-previous-tests). I have flattened the contexts to make it easier to work out what is going on and also to make sure that each test can be run on its own (ie with `it.only`). This means that new tests do not need to depend on the long sequence of test steps given previously. This article explains some of the reasons that we should avoid context nesting in tests: https://kentcdodds.com/blog/avoid-nesting-when-youre-testing

Adds tests for:
- when DnB search fails
- successful DnB search
- countries appear in the overseas countries list
- when DnB company creation fails
- when DnB investigation post fails

## Test instructions

Functionally everything should be the same. You could try running some of the refactored tests with `.only` to make sure that they run independently of other tests.

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"

- [ ] Has the branch been rebased to master?
- [ ] Automated tests (Any of the following when applicable: Unit, Functional or End-to-End)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, IE11, Safari)

### CODE REVIEW

File: test/functional/cypress/specs/companies/add-company-spec.js [hunk 1]
```diff
@@ -52,568 +94,426 @@ describe('Add company form', () => {
...
-          }
-        )
-      }
-    )
+    it('should display "Country" selection', () => {
+      cy.get(selectors.companyAdd.form).contains('Country')
+      cy.get(selectors.companyAdd.form).find('select').should('be.visible')
+    })
+
+    it('should include Hong Kong in the "Country" selection', () => {
```
Comment (tomkingston) [inline test/functional/cypress/specs/companies/add-company-spec.js:123]: This one is a new test

File: test/functional/cypress/specs/companies/add-company-spec.js [hunk 2]
```diff
@@ -52,568 +94,426 @@ describe('Add company form', () => {
...
+      cy.get(selectors.companyAdd.entitySearch.companyNameField).type(
+        'Simulate 500 Error'
+      )
+      cy.get(selectors.companyAdd.entitySearch.searchButton).click()
+      cy.get('[data-test="status-message"]')
+        .should('exist')
+        .should('contain.text', 'Error occurred while searching for company.')
+    })
+  })
+
```
Comment (tomkingston) [inline test/functional/cypress/specs/companies/add-company-spec.js:192]: This is a new test

File: test/functional/cypress/specs/companies/add-company-spec.js [hunk 3]
```diff
@@ -52,568 +94,442 @@ describe('Add company form', () => {
...
+      cy.get(selectors.companyAdd.form).find('select').should('be.visible')
+    })
+
+    it('should include Hong Kong in the "Country" selection', () => {
+      cy.get(selectors.companyAdd.form).contains('Country')
+      cy.get(selectors.companyAdd.form)
+        .find('select option[value="HK"]')
+        .should('have.length', 1)
+        .should('have.text', 'Hong Kong (SAR)')
+    })
```
Comment (tomkingston) [inline test/functional/cypress/specs/companies/add-company-spec.js:129]: This one is a new test

File: test/functional/cypress/specs/companies/add-company-spec.js [hunk 4]
```diff
@@ -52,568 +94,442 @@ describe('Add company form', () => {
...
+    it('should display an error message when DnB search fails', () => {
+      cy.get(selectors.companyAdd.entitySearch.companyNameField).type(
+        'Simulate 500 Error'
+      )
+      cy.get(selectors.companyAdd.entitySearch.searchButton).click()
+      cy.get('[data-test="status-message"]')
+        .should('exist')
+        .should('contain.text', 'Error occurred while searching for company.')
+    })
+  })
```
Comment (tomkingston) [inline test/functional/cypress/specs/companies/add-company-spec.js:191]: This one is a new test

File: test/functional/cypress/specs/companies/add-company-spec.js [hunk 5]
```diff
@@ -52,568 +94,442 @@ describe('Add company form', () => {
...
-      })
-    }
-  )
+    it('should not continue when DnB company creation fails', () => {
+      // Choosing the water sector simulates a server error
+      cy.get(selectors.companyAdd.sectorSelect).select('Water')
+      cy.get(selectors.companyAdd.submitButton).click()
+      cy.location('pathname').should('eq', urls.companies.create())
+    })
+  })
```
Comment (tomkingston) [inline test/functional/cypress/specs/companies/add-company-spec.js:296]: This one is a new test

File: test/functional/cypress/specs/companies/add-company-spec.js [hunk 6]
```diff
@@ -52,568 +94,442 @@ describe('Add company form', () => {
...
+    })
+
+    it('should not continue when the DnB investigation api call fails', () => {
+      cy.get(selectors.companyAdd.newCompanyRecordForm.region).select('London')
+      // Water sector simulates a server error from dnb investigation post
+      cy.get(selectors.companyAdd.newCompanyRecordForm.sector).select('Water')
+      cy.get(selectors.companyAdd.submitButton).click()
+
+      cy.location('pathname').should('eq', urls.companies.create())
+    })
```
Comment (tomkingston) [inline test/functional/cypress/specs/companies/add-company-spec.js:447]: This one is a new test

**Review by cgsunkel — APPROVED** [review]

**Review by ian-leggett — APPROVED** [review]
LGTM

**Pull request merged**